server/sched.go

```diff
@@ -81,6 +90,8 @@ func (s *Scheduler) GetRunner(c context.Context, model *Model, opts api.Options,
 		successCh:       make(chan *runnerRef),
 		errCh:           make(chan error, 1),
 	}
+	// context split across parallel threads
+	opts.NumCtx = opts.NumCtx * numParallel
```

Comment:
I might not be interpreting this incorrectly, but this looks like it assumes both models running in parallel have the same context size, which isn't going to be true a lot of the time.

Reply:
This is one model, many parallel requests. llama.cpp will split the kv cache between all parallel requests so we need to make the context size n times bigger

Reply:
I see, thanks for the clarification